earth_enterprise/src/common/serverdb/serverdbReader.cpp

```diff
@@ -395,9 +443,21 @@ const MimeType ServerdbReader::GetData(
       }
     } else if (request == "Json") {
       std::string json_variable_name = arg_map["var"];
-      // CheckJavascriptName(json_variable_name);
-      GetJson(json_variable_name, language, region, buf);
-      return MapTileUtils::kJavascriptMimeType;
+      if (!arg_map["v"].empty()) {
+        // CheckJavascriptName(json_variable_name);
+        if (arg_map["v"] == "2") {
+          GetJson(json_variable_name, language, region, buf);
+        } else if (arg_map["v"] == "1") {
+          // This is added for backwards compatibility reasons.
+          GetJsonV1(json_variable_name, language, region, buf);
+        } else {
+          throw khSimpleException("Illegal argument found in v parameter: ") << arg_map["v"];
```

Comment:
How throwing this exception affect the calling code ? will it be caught and eventually the user see it ?

Reply:
yes it is caught.  Looking at other code in the module it seems to be the way invalid parameters are handled.  See my testing comment below.